apps/go-mod-validator/test/deps.test.ts

```diff
@@ -0,0 +1,44 @@
+import { execSync } from "child_process";
+import { getAllGoModDeps } from "../src/deps";
+import { describe, expect, it, vi, MockedObject } from "vitest";
+import * as glob from "@actions/glob";
+
+const mockedExecSync = vi.mocked(execSync);
+vi.mock("child_process");
+const mockedGlob = vi.mocked(glob);
+vi.mock("@actions/glob");
+type MockedGlob = MockedObject<Awaited<ReturnType<typeof glob.create>>>;
+
+describe("getDependenciesMap", () => {
+  it("should return a map of <go.mod files: dependencies in json>", async () => {
```

Comment:
Discussed offline, we should think about switching these to actual `go.mod` files rather than mocks, can be done separately though.

Reply:
https://smartcontract-it.atlassian.net/jira/software/c/projects/RE/boards/83?selectedIssue=RE-2792